bika/lims/browser/widgets/analysisprofileanalyseswidget.py

```diff
@@ -48,118 +48,117 @@ def __init__(self, context, request, fieldvalue=[], allow_edit=False):
         member = mtool.getAuthenticatedMember()
         roles = member.getRoles()
-        self.allow_edit = 'LabManager' in roles or 'Manager' in roles
-
-        items = BikaListingView.folderitems(self)
-
-        for x in range(len(items)):
-            if not items[x].has_key('obj'): continue
-            obj = items[x]['obj']
-
-            cat = obj.getCategoryTitle()
-            # Category (upper C) is for display column value
-            items[x]['Category'] = cat
-            if self.do_cats:
-                # category is for bika_listing to groups entries
-                items[x]['category'] = cat
-                if cat not in self.categories:
-                    self.categories.append(cat)
-
-            analyses = [a.UID() for a in self.fieldvalue]
-
-            items[x]['selected'] = items[x]['uid'] in analyses
-
-            items[x]['class']['Title'] = 'service_title'
-
-            calculation = obj.getCalculation()
-            items[x]['Calculation'] = calculation and calculation.Title()
-
-            locale = locales.getLocale('en')
-            currency = self.context.bika_setup.getCurrency()
-            symbol = locale.numbers.currencies[currency].symbol
-            items[x]['Price'] = "%s %s" % (symbol, obj.getPrice())
-            items[x]['class']['Price'] = 'nowrap'
-
-            after_icons = ''
-            if obj.getAccredited():
-                after_icons += "<img\
-                src='%s/++resource++bika.lims.images/accredited.png'\
-                title='%s'>"%(self.context.absolute_url(),
-                              _("Accredited"))
-            if obj.getReportDryMatter():
-                after_icons += "<img\
-                src='%s/++resource++bika.lims.images/dry.png'\
-                title='%s'>"%(self.context.absolute_url(),
-                              _("Can be reported as dry matter"))
-            if obj.getAttachmentOption() == 'r':
-                after_icons += "<img\
-                src='%s/++resource++bika.lims.images/attach_reqd.png'\
-                title='%s'>"%(self.context.absolute_url(),
-                              _("Attachment required"))
-            if obj.getAttachmentOption() == 'n':
-                after_icons += "<img\
-                src='%s/++resource++bika.lims.images/attach_no.png'\
-                title='%s'>"%(self.context.absolute_url(),
-                              _('Attachment not permitted'))
-            if after_icons:
-                items[x]['after']['Title'] = after_icons
-
-            if self.profile:
-                # Display analyses for this Analysis Service in results?
-                ser = self.profile.getAnalysisServiceSettings(obj.UID())
-                items[x]['allow_edit'] = ['Hidden', ]
-                items[x]['Hidden'] = ser.get('hidden', obj.getHidden())
-
+        self.allow_edit = "LabManager" in roles or "Manager" in roles
         self.categories.sort()
```

Comment:
I think this `self.categories.sort()` does nothing here in `before_render`. Since we don't have an `after_render` function (that would be nice), the best option would be to add the `folderitems` function, call super and sort the categories thereafter. Is not elegant, I know, but at least the Analysis Categories will appear sorted alphabetically when "Categorize Analysis Services" is enabled. With this code, the categories will appear in the order the analyses are retrieved (by `sortable_title`) which is not bad, but is not consistent with the rest.

Reply:
good catch, thanks, I'll change that